collective-project-backend/Controllers/StudentController.cs

```diff
@@ -7,25 +7,162 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Services;
+using DatabaseAccess.Models;
 
 namespace API.Controllers
 {
-    [Route("students")]
-    [ApiController]
-    [Authorize]
-    public class StudentController : ControllerBase
-    {
-        private readonly StudentService _studentService;
-
-        public StudentController(StudentService studentService)
-        {
-            _studentService = studentService;
-        }
-
-        [HttpGet]
-        public ActionResult<IList<Student>> GetAllStudents()
-        {
-            return Ok(_studentService.GetAllStudents());
-        }
-    }
+	[Route("students")]
+	[ApiController]
+	[Authorize]
+	public class StudentController : ControllerBase
+	{
+		private readonly StudentService _studentService;
+		private readonly ApplicationService _applicationService;
+		private readonly InternshipService _internshipService;
+		private readonly SubscriptionService _subscriptionService;
+
+		public StudentController(StudentService studentService, ApplicationService applicationService,
+			InternshipService internshipService, SubscriptionService subscriptionService)
+		{
+			_studentService = studentService;
+			_applicationService = applicationService;
+			_internshipService = internshipService;
+			_subscriptionService = subscriptionService;
+		}
+
+		[HttpGet("getStudents")]
+		public IActionResult GetAllStudents()
+		{
+			return Ok(_studentService.GetAllStudents());
+		}
+
+		[HttpGet("getStudentsByInternshipId")]
+		public IActionResult GetStudentsByInternshipId(int InternshipId)
+		{
+			//var InternshipId = Request.Cookies["InternshipId"];
+			var studentIds = _applicationService.GetStudentIdsByInternshipId(InternshipId);
+			var students = _studentService.GetAllStudents();
+			IList<Student> studentsByInternshipId = new List<Student>();
+			foreach (Student student in students)
+			{
+				for (int i = 0; i < studentIds.Count(); i++)
+				{
+					if (student.Id == studentIds[i])
+					{
+						studentsByInternshipId.Add(student);
+					}
+				}
+			}
+			return Ok(studentsByInternshipId);
+		}
+
+		[HttpGet("GetStudentsByCompanyId")]
+		public IActionResult GetStudentsByCompanyId(int CompanyId)
+		{
+			//var CompanyId = Request.Cookies["CompanyId"];
+			var studentIds = _subscriptionService.GetStudentIdsByCompanyId(CompanyId);
+			var students = _studentService.GetAllStudents();
+			IList<Student> studentsByCompanyId = new List<Student>();
+			foreach (Student student in students)
+			{
+				for (int i = 0; i < studentIds.Count(); i++)
+				{
+					if (student.Id == studentIds[i])
+					{
+						studentsByCompanyId.Add(student);
+					}
+				}
+			}
+			return Ok(studentsByCompanyId);
+		}
+
+		[HttpPost("addStudent")]
+		public IActionResult AddStudent([FromBody] Student student)
+		{
+			_studentService.AddStudent(student);
+			return Ok();
+		}
+
+		[HttpPost("updateStudent")]
+		public IActionResult UpdateStudent([FromBody] Student student)
```

Comment:
obiectul primit in body ar trebui sa fie un ViewModel (care sa contina doar fieldurile ce pot fi updatate), nu tot obiectul din db
ar trebui adaugat rolul care poate face update unui student